File: video.java
public class Video {
    public int Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public string FilePath { get; set; }
    public string YouTubeUrl { get; set; }
    public DateTime UploadDate { get; set; }
}
public class VideoRequest {
    public string Title { get; set; }
    public string Description { get; set; }
    public string Type { get; set; }
    public string Url { get; set; }
    public string Thumbnail { get; set; }
}
public class VideoResponse {
    public string Message { get; set; }
    public Video Video { get; set; }
}
const string videoId = "1234567890";

var video = await _videoService.GetVideo(videoId);
if (video == null) {
    return NotFound();
}

return Ok(video);

File: video.ts
const port = 3000;

type VideoType = 'movie' | 'music' | 'tutorial' | 'other';

interface Video {
    title: string;
    description: string;
    type: VideoType;
    url: string;
    thumbnail: string;
}

interface VideoRequest {
    title: string;
    description: string;
    type: VideoType;
    url: string;
    thumbnail: string;
}

interface VideoResponse {
    message: string;
    video: Video;
}

interface ErrorResponse {
    message: string;
}

interface SearchResponse {
    message: string;
    videos: Video[];
}

interface SearchRequest {
    query: string;
}

interface UploadResponse {
    message: string;
    video: Video;
}

interface UploadRequest {
    title: string;
    description: string;
    type: VideoType;
    url: string;
    thumbnail: string;
}

interface DownloadResponse {
    message: string;
    video: Video;
}

interface DownloadRequest {
    videoUrl: string;
    title: string;
}

interface DeleteResponse {
    message: string;
}

interface DeleteRequest {
    videoUrl: string;
}

interface LikeResponse {
    message: string;
}

interface LikeRequest {
    videoUrl: string;
}

interface WatchLaterResponse {
    message: string;
}

interface WatchLaterRequest {
    videoUrl: string;
}
